### PR TITLE
3/177 hotbar implementation

### PR DESCRIPTION
**Pull request by Pratulw5**

### CODE REVIEW

**Review by ishusharma13 — APPROVED** [review]
Seems good!

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java [hunk 1]
```diff
@@ -34,9 +34,11 @@ public class PlayerInventoryDisplay extends UIComponent {
     private boolean toggle = false; // Whether inventory is toggled on;
     DialogueBox itemOverlay;
 
+    private boolean toggleHotbar = true;
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java:37]: Is there a reason we need a toggleHotbar, should this not just always be the opposite of toggle?

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java [hunk 2]
```diff
@@ -96,6 +100,11 @@ private void toggleInventory() {
         }
     }
 
+    public void toggleHotbar() {
+        if (toggleHotbar) hotbar.disposeTable();
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java:104]: Just use toggle, no need for toggleHotbar

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java [hunk 3]
```diff
@@ -3,135 +3,116 @@
...
+     *
+     * @param hotBarCapacity Number of slots in the hotbar
+     * @param inventory      Player's inventory
+     * @param playerInventoryDisplay Inventory display manager
+     */
     public PlayerInventoryHotbarDisplay(int hotBarCapacity, Inventory inventory, PlayerInventoryDisplay playerInventoryDisplay) {
-        if (hotBarCapacity < 1) {
-            throw new IllegalArgumentException("Inventory Hotbar dimensions must be more than one!");
-        }
+        if (hotBarCapacity < 1) throw new IllegalArgumentException("Inventory Hotbar dimensions must be more than one!");
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java:41]: just call this capacity (also throw in a check to ensure capacity <= inventory.getCapacity())

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java [hunk 4]
```diff
@@ -3,135 +3,116 @@
...
         this.inventory = inventory;
-        this.hotBarTexture=new Texture("Inventory/hotbar.png");
-        this.playerInventoryDisplay=playerInventoryDisplay;
+        this.hotBarCapacity=hotBarCapacity;
+        this.playerInventoryDisplay = playerInventoryDisplay;
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java:44]: again, no need to call this playerInventoryDisplay (no Hungarian notation) - inventoryDisplay or inventoryUI is enough

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java [hunk 5]
```diff
@@ -34,9 +34,11 @@ public class PlayerInventoryDisplay extends UIComponent {
     private boolean toggle = false; // Whether inventory is toggled on;
     DialogueBox itemOverlay;
 
+    private boolean toggleHotbar = true;
 
-    private final Skin skininv=new Skin(Gdx.files.internal("Inventory/inventory.json"));
-    private final Skin skinSlots=new Skin(Gdx.files.internal("Inventory/skinforslot.json"));
+
+    private final Skin skininv=new Skin(Gdx.files.internal("Inventory/inventory.json"));//created by @PratulW5
+    private final Skin skinSlots=new Skin(Gdx.files.internal("Inventory/skinforslot.json"));//created by @PratulW5
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java:42]: Adding slot listeners should not be a void function (ie anyone can access it). At least make it public void

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java [hunk 6]
```diff
@@ -3,135 +3,116 @@
...
-import com.badlogic.gdx.scenes.scene2d.InputEvent;
-import com.badlogic.gdx.scenes.scene2d.InputListener;
 import com.badlogic.gdx.scenes.scene2d.ui.*;
-import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
-import com.badlogic.gdx.scenes.scene2d.utils.Drawable;
+
 import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
 import com.csse3200.game.inventory.Inventory;
 import com.csse3200.game.inventory.items.AbstractItem;
 import com.csse3200.game.services.ServiceLocator;
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java:11]: Fix up spacing in files (no need for horizontal whitespace everywhere).

**Review by mitsudz — CHANGES_REQUESTED** [review]

File: saves/quests.json [hunk 7]
```diff
@@ -1,68 +1,68 @@
...
-				]
-			}
-			isSecretQuest: false
-			questDialogue: null
-			currentTaskIndex: 1
-			isFailed: false
-			isActive: false
-			taskCompletionTriggers: null
+quests: [
+	{
```
Comment (mitsudz) [inline saves/quests.json:3]: Should not have changes to saves files

**Review by mitsudz — COMMENTED** [review]
There shouldn't be any changed files other than PlayerInventoryDisplay and PlayerHotbarDisplay?

**Review by mitsudz — APPROVED** [review]

**Pull request merged**